pkg/templates/antiaffinity/template.go

```diff
@@ -59,30 +61,47 @@ func init() {
 				if !hasPods {
 					return nil
 				}
+				var foundIssues []diagnostic.Diagnostic
 				if affinity := podTemplateSpec.Spec.Affinity; affinity != nil && affinity.PodAntiAffinity != nil {
 					preferredAffinity := affinity.PodAntiAffinity.PreferredDuringSchedulingIgnoredDuringExecution
 					requiredAffinity := affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution
 					for _, preferred := range preferredAffinity {
-						if affinityTermMatchesLabelsAgainstNodes(preferred.PodAffinityTerm, podTemplateSpec.Namespace, podTemplateSpec.Labels, topologyKeyMatcher) {
+						err := affinityTermMatchesLabelsAgainstNodes(preferred.PodAffinityTerm,
+							podTemplateSpec.Namespace, podTemplateSpec.Labels, topologyKeyMatcher)
+						if err == nil {
 							return nil
 						}
+						foundIssues = append(foundIssues, diagnostic.Diagnostic{
+							Message: err.Error(),
+						})
 					}
 					for _, required := range requiredAffinity {
-						if affinityTermMatchesLabelsAgainstNodes(required, podTemplateSpec.Namespace, podTemplateSpec.Labels, topologyKeyMatcher) {
+						err := affinityTermMatchesLabelsAgainstNodes(required, podTemplateSpec.Namespace,
+							podTemplateSpec.Labels, topologyKeyMatcher)
+						if err == nil {
 							return nil
 						}
+						foundIssues = append(foundIssues, diagnostic.Diagnostic{
+							Message: err.Error(),
+						})
 					}
 				}
+				if foundIssues != nil {
```

Comment:
I think this should be in above `if` body

Reply:
Now this condition is redundant

Reply:
![Screenshot 2022-05-05 at 18 06 23](https://user-images.githubusercontent.com/89904305/166965559-8972838c-48d3-45ee-b01e-c90beda8c9ab.png)